### DISCUSSION TITLE
Cannot use rdkit-js easily in NextJS - example limitation?

### DISCUSSION THREAD

**Discussion by sgoodrow-zymergen** [rdkit/rdkit-js#360] [General]
Hi,

We are using `rdkit-js` for molecule image rendering. Thank you for the great library and work, we appreciate it.

I am attempting to use the library in a NextJS app. We happen to be using the new "app router" architecture, but I think this issue will exist in the legacy "pages router" architecture as well.

**I am trying to use it on the server-side, not in the browser (which I have gotten to work by using the unpkg source in a script tag).**

The `rdkit-js` loading script seems to look for the `.wasm` file relative to its loading route. This means that if the loading script is called from a particular nested NextJS route path, the `.wasm` file has to be at that path location as well.

This is not seen in the example NextJS app because the loading script is always called from `/`, so it is expected to be found at `/chunks/app`, but if you _also_ load the script from `/something`, it will attempt to look for the`.wasm` file in `/chunks/app/something`.

While it is straight-forward to copy the `.wasm` file into that location instead, I'd really like to only ship the `.wasm` file once and be able to use rdkit from any path.

**Comment by ptosco** [comment]
You may call `initRDKitModule` as follows:
```javascript
initRDKitModule({
    locateFile: () => '/path/to/RDKIt_minimal.wasm'
});
```
where `/path/to` is the location of the `.wasm` file on your server.
This should solve your problem.

**Reply by sgoodrow-zymergen** [reply]
Oh cool! I must've missed that when reviewing the documentation. I'll try that and report back thanks. 🙂

Would you like a PR with that updated in the type definition? https://github.com/rdkit/rdkit-js/blob/master/typescript/index.d.ts#L538

**Comment by ptosco** [comment]
I do not use TypeScript myself, but I am pretty sure @MichelML would be happy to review and eventually merge your PR :-)

**Comment by MichelML** [comment]
@ptosco thanks for this I also didn't know!

@sgoodrow-zymergen yes a PR would be appreciated, but also if you could specify this option in the main readme, I think it could help other people to be rapidly aware of this option when using rdkit-js in similar contexts as yours!

**Comment by sgoodrow-zymergen** [comment]
Apologies for the delay, here's the PR: https://github.com/rdkit/rdkit-js/pull/362

Thanks!
